hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go

```diff
@@ -640,6 +640,17 @@ func (r *DedicatedServingComponentSchedulerAndSizer) updateHostedCluster(ctx con
 	if sizeConfig.Effects != nil && sizeConfig.Effects.APICriticalPriorityClassName != nil {
 		hc.Annotations[hyperv1.APICriticalPriorityClass] = *sizeConfig.Effects.APICriticalPriorityClassName
 	}
+	if sizeConfig.Effects != nil && sizeConfig.Effects.MachineHealthCheckTimeout != nil {
+		hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation] = sizeConfig.Effects.MachineHealthCheckTimeout.Duration.String()
+	} else {
+		// If mhc timeout is configured for any size in the config, remove the annotation
+		// to fallback to the default
+		if configHasMHCTimeout(config) {
```

Comment:
why this check, should we just delete it?

Reply:
The reasoning was that if you had configured a MHC timeout anywhere in your config, we would remove the annotation here. Otherwise, leave it alone in case (for whatever reason) you're simply annotating the HostedCluster.